include/pinocchio/algorithm/regressor.hpp

```diff
@@ -372,6 +372,61 @@ namespace pinocchio
     const ModelTpl<Scalar, Options, JointCollectionTpl> & model,
     DataTpl<Scalar, Options, JointCollectionTpl> & data,
     const Eigen::MatrixBase<ConfigVectorType> & q);
+
+  ///
+  /// \brief Computes the momentum regressor and component of time derivative
+  /// of momentum regressor of the system based on the current robot state.
+  ///
+  /// The result stored in `data.momentumRegressor` corresponds to a matrix \f$ Y_p \f$ such that
+  /// \f$ P = Y_p(q, v) \pi = M v \f$ where \f$ \pi \f$ represents the vector of dynamic parameters
+  /// of each link.
+  ///
+  /// The result stored in `data.dpartial_lagrangian_q` corresponds to a matrix
+  /// \f$ Y_h \f$ such that \f$ \frac{\partial L}{\partial q} = Y_h(q, v) \f$.
+  ///
+  /// \tparam JointCollection Collection of Joint types.
+  /// \tparam ConfigVectorType Type of the joint configuration vector.
+  ///
+  /// \param[in] model The model structure representing the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[in] q The joint configuration vector (dim model.nq).
+  /// \param[in] v The joint velocity vector (dim model.nv).
+  ///
+  /// \return A pair containing:
+  ///         - The momentum regressor matrix.
+  ///         - A matrix containing a component of the time derivative of the momentum regressor.
+  ///
+  template<
+    typename Scalar,
+    int Options,
+    template<typename, int>
+    class JointCollectionTpl,
+    typename ConfigVectorType,
+    typename TangentVectorType>
+  std::pair<
+    typename DataTpl<Scalar, Options, JointCollectionTpl>::MatrixXs,
+    typename DataTpl<Scalar, Options, JointCollectionTpl>::MatrixXs>
+  computeMomentumRegressor(
+    const ModelTpl<Scalar, Options, JointCollectionTpl> & model,
+    DataTpl<Scalar, Options, JointCollectionTpl> & data,
+    const Eigen::MatrixBase<ConfigVectorType> & q,
+    const Eigen::MatrixBase<TangentVectorType> & v);
+
+  template<
```

Comment:
Hi, could you please add a comment to explain the mathematical meaning of computeIndirectRegressors?

Reply:
I think there is no mathematical meaning behind. It is just that the function in Wensing's implementation had this comment:

```matlab
function  [Y_Hqd, Y_CTqd, Y_Hqd_rot, Y_CTqd_rot]  = Regressor_HqdandCTqd( model, q , qd)
% Indirect regressors term  YHqd a = H qd and YCTqd = C^T qd
```